opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java

```diff
@@ -847,25 +848,31 @@ public void setProjects(Map<String, Project> projects) {
     }
 
     /**
-     * Adds a group to the set. This is performed upon configuration parsing
+     * Adds a group to the map. This is performed upon configuration parsing
      *
      * @param group group
      * @throws IOException when group is not unique across the set
      */
     public void addGroup(Group group) throws IOException {
-        if (!groups.add(group)) {
+        if(!groups.containsKey(group.getName())){
             throw new IOException(
                     String.format("Duplicate group name '%s' in configuration.",
                             group.getName()));
-        }
+            }else{
+                groups.put(group.getName(), group);
+            }
     }
 
-    public Set<Group> getGroups() {
+    public Map<String, Group> getGroups() {
         return groups;
     }
 
-    public void setGroups(Set<Group> groups) {
-        this.groups = groups;
+    public void setGroups(Map<String, Group> groups) {
+        Map<String, Group> settedGroup = new HashMap<String, Group>();
```

Comment:
In constructor, there is a `TreeMap` so I'd assume it was used for a reason. If tests pass, then using `HashMap` is okay; however, please use "copy constructor" like: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/TreeMap.html#%3Cinit%3E(java.util.Map)

Reply:
Originally a treeset was used so that the groups are ordered on the main page. This needs be respected.

Reply:
the original code of this method does not do a copy, why do you need it now?